### ISSUE TITLE
Incorrect method annotation is misleading

### ISSUE THREAD

**Issue by brendanWeb** [magento/magento2#7637]
### Preconditions

1. Magento version 2.1.2 (Also checked against HEAD of develop branch)
2. MySQL 5.7
3. PHP 7.0.12

### Steps to reproduce

1. Create an ```InstallSchema``` or ```UpgradeSchema``` class in a custom module and add the following line of code somewhere within:

```
$setup->getConnection->addColumn('testing',
    'name',
    [
        'TYPE' => \Magento\Framework\DB\Ddl\Table::TYPE_TEXT,
        'NULLABLE' => true,
        'COMMENT' => 'Name'
    ]
)->commit();
```

### Expected result

1. For the specific error mentioned below to not occur

### Actual result

1. A fatal PHP error describing that there is no ```commit()``` method implemented for type ```Magento\Framework\DB\Statement\Pdo\Mysql``` which extends the ```\Zend_Db_Statement_Pdo``` class shown in the correct annotation

The above error occurs because:

The ```@return``` annotation on the ```addColumn()``` method of the ```\Magento\Framework\DB\Adapter\AdapterInterface``` is:

```
* @return \Magento\Framework\DB\Adapter\AdapterInterface
```

This can be confusing for a developer and is technically incorrect because:

The only implementation of that interface(```\Magento\Framework\DB\Adapter\Pdo\Mysql```) shows the following ```@return``` annotation for the ```addColumn()``` method:

```
* @return  true|\Zend_Db_Statement_Pdo
```

In summary, the value that is returned from the ```addColumn()``` method is consistent with the annotation on the concrete implementation of the method. This can be misleading for developers who depend on the interface as they would be led to believe that you could do something such as what was shown in the **Steps to reproduce** section.

I have created a pull request to address this minor change.

**Comment by magento-engcom-team** [comment]
@brendanWeb, thank you for your report.
We've created internal ticket(s) MAGETWO-81738 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @lisovyievhenii. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.
